### PR TITLE
APIS-5785 Add new application state PENDING_RESPONSIBLE_INDIVIDUAL_VERIFICATION

### PR DESCRIPTION
**Pull request by petekirby-ee**

### CODE REVIEW

File: app/uk/gov/hmrc/apiplatform/modules/approvals/services/ResponsibleIndividualVerificationService.scala [hunk 1]
```diff
@@ -51,25 +61,49 @@ class ResponsibleIndividualVerificationService @Inject()(
   }
 
   def accept(code: String): Future[Either[String, ResponsibleIndividualVerification]] = {
+
+    def getResponsibleIndividualEmailFromStdApp(std: Standard): Option[String] = {
+      std.importantSubmissionData.fold[Option[String]](None)(isd => Some(isd.responsibleIndividual.emailAddress.value))
```
Comment (AndySpaven) [inline app/uk/gov/hmrc/apiplatform/modules/approvals/services/ResponsibleIndividualVerificationService.scala:66]: Just `.map` - no need for fold

File: app/uk/gov/hmrc/apiplatform/modules/approvals/services/ResponsibleIndividualVerificationService.scala [hunk 2]
```diff
@@ -97,4 +131,18 @@ class ResponsibleIndividualVerificationService @Inject()(
       } yield riVerification
     ).value
   }
+
+  private def writeStateHistory(snapshotApp: ApplicationData, name: String) = 
+    insertStateHistory(snapshotApp, PENDING_GATEKEEPER_APPROVAL, Some(PENDING_RESPONSIBLE_INDIVIDUAL_VERIFICATION), name, COLLABORATOR, (a: ApplicationData) => applicationRepository.save(a))
+
+  private def insertStateHistory(snapshotApp: ApplicationData, newState: State, oldState: Option[State],
```
Comment (AndySpaven) [inline app/uk/gov/hmrc/apiplatform/modules/approvals/services/ResponsibleIndividualVerificationService.scala:138]: This probably should not be duplicated in the 3 places (2 before you started).  Maybe.

File: app/uk/gov/hmrc/thirdpartyapplication/domain/models/ApplicationState.scala [hunk 3]
```diff
@@ -65,6 +66,21 @@ case class ApplicationState(
       requestedByEmailAddress = Some(requestedByEmailAddress))
   }
 
+  def toPendingGatekeeperApproval(clock: Clock) = {
+    requireState(requirement = PENDING_RESPONSIBLE_INDIVIDUAL_VERIFICATION, transitionTo = State.PENDING_GATEKEEPER_APPROVAL)
```
[the remark this answers is not in the document]
Comment (petekirby-ee) [reply app/uk/gov/hmrc/thirdpartyapplication/domain/models/ApplicationState.scala:70]: That's immediately above, with the requestedByEmailAddress

File: app/uk/gov/hmrc/apiplatform/modules/approvals/services/ResponsibleIndividualVerificationService.scala (hunk 1 above)
Comment (petekirby-ee) [reply app/uk/gov/hmrc/apiplatform/modules/approvals/services/ResponsibleIndividualVerificationService.scala]: Fixed

File: app/uk/gov/hmrc/apiplatform/modules/approvals/services/ResponsibleIndividualVerificationService.scala (hunk 2 above)
Comment (petekirby-ee) [reply app/uk/gov/hmrc/apiplatform/modules/approvals/services/ResponsibleIndividualVerificationService.scala]: Moved method insertStateHistory into new base class

**Pull request merged**